packages/client/hmi-client/src/workflow/ops/simulate-julia/tera-simulate-node-julia.vue

```diff
@@ -161,33 +191,42 @@ watch(
 
 watch(() => completedRunIdList.value, watchCompletedRunList, { immediate: true });
 
-onMounted(async () => {
-	const node = props.node;
-	if (!node) return;
+const lazyLoadRunResults = async (runId: string) => {
+	if (runResults.value[runId]) return;
 
-	const port = node.outputs[0];
-	if (!port) return;
+	const resultCsv = await getRunResult(runId, 'result.csv');
+	const csvData = csvParse(resultCsv);
 
-	const runIdList = port.value as string[];
-	await Promise.all(
-		runIdList.map(async (runId) => {
-			const resultCsv = await getRunResult(runId, 'result.csv');
-			const csvData = csvParse(resultCsv);
-
-			const configId = props.node.inputs[0].value?.[0];
-			if (configId) {
-				const modelConfig = await getModelConfigurationById(configId);
-				const parameters = modelConfig.configuration.semantics.ode.parameters;
-				csvData.forEach((row) =>
-					parameters.forEach((parameter) => {
-						row[parameter.id] = parameter.value;
-					})
-				);
-			}
-			runResults.value[runId] = csvData as any;
-		})
-	);
-});
+	const configId = props.node.inputs[0].value?.[0];
+	if (configId) {
+		const modelConfig = await getModelConfigurationById(configId);
+		const parameters = modelConfig.configuration.semantics.ode.parameters;
+		csvData.forEach((row) =>
+			parameters.forEach((parameter) => {
+				row[parameter.id] = parameter.value;
+			})
+		);
```

Comment:
What makes you assume that `csvData` and `ode.parameters` will be in the same order?

Reply:
marking as resolved for now as we'll take a deeper look at this in the future as per our conversation earlier